src/sentry/static/sentry/app/components/search/sources/helpSource.tsx

```diff
@@ -0,0 +1,182 @@
+import React from 'react';
+import debounce from 'lodash/debounce';
+import dompurify from 'dompurify';
+import {withRouter, WithRouterProps} from 'react-router';
+import {
+  SentryGlobalSearch,
+  standardSDKSlug,
+  Result as SearchResult,
+} from '@sentry-internal/global-search';
+
+import withLatestContext from 'app/utils/withLatestContext';
+import {Organization, Project} from 'app/types';
+import parseHtmlMarks from 'app/utils/parseHtmlMarks';
+
+type MarkedText = ReturnType<typeof parseHtmlMarks>;
+
+type ResultItem = {
+  sourceType: 'help';
+  resultType: string;
+  title: string;
+  description?: string;
+  to?: string;
+  /**
+   * Context will be mapped into the extra node
+   */
+  extra?: string;
+  /**
+   * Section heading is declared when the first result designates a section of the
+   * global search results.
+   */
+  sectionHeading?: string;
+  sectionCount?: number;
+  empty?: boolean;
+} & (
+  | SearchResult['hits'][0]
+  | {
+      /**
+       * When we have no results for a section we mark the result item as empty
+       */
+      empty?: true;
+    }
+);
+
+type Result = {
+  item: ResultItem;
+  matches?: MarkedText[];
+};
+
+type RenderProps = {
+  isLoading: boolean;
+  /**
+   * Matched results
+   */
+  results: Result[];
+  allResults: Result[];
+};
+
+type Props = WithRouterProps & {
+  organization: Organization;
+  project: Project;
+  /**
+   * Specific platforms to filter reults to
+   */
+  platforms: string[];
+  /**
+   * The string to search the navigation routes for
+   */
+  query: string;
+  /**
+   * Render function that renders the global search result
+   */
+  children: (props: RenderProps) => React.ReactNode;
+};
+
+type State = {
+  loading: boolean;
+  results: Result[];
+};
+
+const MARK_TAGS = {
+  highlightPreTag: '<mark>',
+  highlightPostTag: '</mark>',
+};
+
+class HelpSource extends React.Component<Props, State> {
+  state: State = {
+    loading: false,
+    results: [],
+  };
+
+  componentDidMount() {
+    if (this.props.query !== undefined) {
```

Comment:
do we also want to do a search when the string is empty?

Reply:
no but this is fine, it won't do an empty search because of configuration higher up